packages/@o3r/core/schematics/component/common/analytics.ts

```diff
@@ -0,0 +1,74 @@
+import { externalSchematic, Rule, SchematicContext } from '@angular-devkit/schematics';
+import { askConfirmation, askQuestion } from '@angular/cli/src/utilities/prompt';
+import { setupSchematicsDefaultParams } from '@o3r/schematics';
+import { NgGenerateComponentSchematicsSchema } from '../schema';
+
+const analyticsPackageName = '@o3r/analytics';
+const addAnalyticsSchematicName = 'analytics-to-component';
+
+export const getAddAnalyticsRules = async (
+  componentPath: string,
+  options: Pick<NgGenerateComponentSchematicsSchema, 'useOtterAnalytics' | 'skipLinter' | 'activateDummy'>,
+  context: SchematicContext
+): Promise<Rule[]> => {
+  let useOtterAnalytics = options.useOtterAnalytics;
+  let alwaysUseOtterAnalytics;
+  try {
+    require.resolve(`${analyticsPackageName}/package.json`);
+    if (useOtterAnalytics === null && context.interactive) {
+      useOtterAnalytics = await askConfirmation('Generate component with Otter analytics?', true);
+      if (useOtterAnalytics) {
+        alwaysUseOtterAnalytics = await askQuestion('Generate future components with Otter analytics by default?', [
```

Comment:
Could make sense to have it shared in the `@o3r/schematics` package with a more generic question
(such as "Use this response as default value?")

Reply:
I will do another PR to refactor a bit all the component schematics code to share a maximum of lines, that we didn't think we can share at the beginning
I can include this fix in this PR